contents/handbook/engineering/product-engineering.md

```diff
@@ -0,0 +1,149 @@
+
+Just shipping makes sense when it's very obviously in line with our company strategy (which is generally proven), and you can descope it successfully. This is almost _everything_ that you may ever build here. The key is to manage the rollout carefully.
+
+Products at PostHog generically go through three phases, and considering your phase is important when you ship new features:
+
+#### 1. Pre PMF at PostHog
+
+* We only build things that _already_ have successful competitors with real revenue. The implication is that "just build it" works disproportionately well because other people have already figured out that new product X has product market fit.
+
+* The challenge is therefore figuring out if _PostHog's_ users base will want the new product, as we already know the product is useful.
+
+#### 2. Figuring out PMF at PostHog
+
+- This is where we are getting our very first paying customers.
+
+- The product focus is making sure people are _delighted_ with the above features. Maybe there are bugs or maybe there are too many gaps with competitors still for people to pay. Focus on how happy users are and why / why not. Keep an eye on early revenue data too - are people willing to pay, or are they churning?
+
+#### 3. Post PMF at PostHog
+
+* This is where we're scaling the number of free and paid users.
+
+* Features at this stage either fall into:
+  * (i) gaps with competitive products that we've not prioritized so far, probably based on feature requests from users, in which case the risk of them not being useful for users is prettty low or...
+  * (ii) totally innovative things like new UX driven by our take on AI, or a new way to access data (like Hog or HogQL), or an integrated experience that no one else can offer because they don't have all the tools in one. In these cases, it is _more important_ to consider how your products are being used as you are more likely to build something that isn't useful (but at this stage, it's fine and encouraged to innovate)
+
+There are plenty of other techniques, that you can do in parallel to get a signal on a new idea:
+
+* **Use the public roadmap.** If you have a rough idea, create a GitHub issue and create an item on posthog.com/roadmap. This will gauge demand.
+
+* **Ask internally for help.** There are lots of people that can help you. The CS team talk to users all the time, the support team have a strong sense of pain points, other product engineers have all talked to users, James and Tim have a broader view of how PostHog is doing, the marketing team can help you get usage or validate demand.
+
+* **Interviews.** Our Product Managers regularly run interviews, ask to be included and give a heads up what you're trying to learn about, or just message users directly! You can even embed your calendar in our surveys product to get ou will need lots of existing potentially relevant customers for this to make sense.
```

Comment:
```suggestion
* **Interviews.** Our Product Managers regularly run interviews, ask to be included and give a heads up what you're trying to learn about, or just message users directly! You can even embed your calendar in our surveys product to book your own user interviews. You will need lots of existing potentially relevant customers for this to make sense, since response rates are typically low.
```